new_structure/views/classteacher.py

```
@@ -3611,114 +3700,136 @@ def get_subjects_by_education_level(education_level):
-            students = Student.query.filter_by(stream_id=stream_obj.id).order_by(Student.name).all()
-
-            # Get subjects for this education level
-            subjects = [subject.name for subject in Subject.query.filter_by(education_level=education_level).all()]
+    try:
+        # Get parameters from the request
+        education_level = request.args.get('education_level', '')
+        grade_level = request.args.get('grade', '')
+        stream_name = request.args.get('stream', '')
+        term = request.args.get('term', '')
+        assessment_type = request.args.get('assessment_type', '')
+        file_format = request.args.get('format', 'xlsx')
```

Ruff (0.12.2)

Local variable `file_format` is assigned to but never used

Remove assignment to unused variable `file_format`

(F841)